src/api/routers/gcp/embeddings.py

```diff
@@ -0,0 +1,122 @@
+import httpx
+import json
+import logging
+import os
+
+from fastapi import APIRouter, Depends, Request, Response
+from api.auth import api_key_auth
+from api.schema import EmbeddingsResponse
+from api.setting import API_ROUTE_PREFIX
+
+from api.modelmapper import get_model
+from api.gcp.credentials.metadata import get_access_token, project_id, location
+
+router = APIRouter(
+    prefix="/embeddings",
+    dependencies=[Depends(api_key_auth)],
+)
+def get_proxy_target(model, path):
+    """
+    Check if the environment variable is set to use GCP.
+    """
+    if os.getenv("PROXY_TARGET"):
+        return os.getenv("PROXY_TARGET")
+    else:
+        return f"https://{location}-aiplatform.googleapis.com/v1/projects/{project_id}/locations/{location}/{model}:predict"
+
+def get_header(model, request, path):
+    path_no_prefix = f"/{path.lstrip('/')}".removeprefix(API_ROUTE_PREFIX)
+    target_url = get_proxy_target(model, path_no_prefix)
+
+    # remove hop-by-hop headers
+    headers = {
+        k: v for k, v in request.headers.items()
+        if k.lower() not in {"host", "content-length", "accept-encoding", "connection", "authorization"}
+    }
+
+    # Fetch service account token
+    access_token = get_access_token()
+    headers["Authorization"] = f"Bearer {access_token}"
+    return target_url, headers
+
+def to_vertex_embeddings(request):
+    """
+    Convert OpenAI-style embeddings request to Vertex AI format.
+    """
+    vertex_request = {
+        "instances": []
+    }
+
+    msg_input = request.get("input")
+    if type(msg_input) is str:
+        vertex_request["instances"] = [{
+            "content": f"{msg_input}"
+            }]
+    elif type(msg_input) is list:
+        vertex_request["instances"] = [{"content": f"{str(item)}"} for item in msg_input]
+
+    return vertex_request
```

Comment:
I think we can tighten this up a bit. What do you think of this?

```suggestion
    inputs = request.get("input", [])
    if not isinstance(inputs, list):
        inputs = [inputs]
    return {
        "instances": [{"content": str(content)} for content in inputs]
    }
```